src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -21,31 +20,39 @@
 class OrderRepository extends CartRepository implements OrderRepositoryInterface
 {
     /**
-     * Create customer orders paginator.
-     *
-     * @param CustomerInterface $customer
-     * @param array             $sorting
-     *
-     * @return Pagerfanta
+     * {@inheritdoc}
      */
     public function createByCustomerPaginator(CustomerInterface $customer, array $sorting = [])
     {
-        $queryBuilder = $this->getCollectionQueryBuilderByCustomer($customer, $sorting);
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        $queryBuilder
+            ->andWhere($queryBuilder->expr()->isNotNull('o.completedAt'))
+            ->innerJoin('o.customer', 'customer')
+            ->andWhere('customer = :customer')
+            ->setParameter('customer', $customer)
+        ;
+
+        $this->applySorting($queryBuilder, $sorting);
 
         return $this->getPaginator($queryBuilder);
     }
 
     /**
-     * Gets orders for customer.
-     *
-     * @param CustomerInterface $customer
-     * @param array             $sorting
-     *
-     * @return array
+     * {@inheritdoc}
      */
     public function findByCustomer(CustomerInterface $customer, array $sorting = [])
     {
-        $queryBuilder = $this->getCollectionQueryBuilderByCustomer($customer, $sorting);
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        $queryBuilder
+            ->andWhere($queryBuilder->expr()->isNotNull('o.completedAt'))
+            ->innerJoin('o.customer', 'customer')
+            ->andWhere('customer = :customer')
+            ->setParameter('customer', $customer)
+        ;
+
+        $this->applySorting($queryBuilder, $sorting);
```

Comment:
Can we extract the QueryBuilder building to a private method? It's duplicated in `findByCustomer` and `createByCustomerPaginator`.